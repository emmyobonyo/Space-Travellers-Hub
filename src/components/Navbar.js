import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../images/planet.png';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Rockets',
    },
    {
      id: 2,
      path: '/missions',
      text: 'Missions',
    },
    {
      id: 3,
      path: '/myprofile',
      text: 'My Profile',
    },
  ];

  return (
    <>
      <nav className="nav-bar">
        <div className="image">
          <img src={logo} className="img" alt="Space-Ship Logo" />
          <h3 className="head">Space Travelers&apos; Hub</h3>
        </div>
        <ul className="list">
          {links.map((link) => (
            <li key={link.id}>
              <NavLink
                to={link.path}
                activeClassName="current"
                className="nav-link"
                exact
              >
                {link.text}
              </NavLink>
            </li>
          ))}
        </ul>
      </nav>
    </>
  );
};

export default Navbar;
