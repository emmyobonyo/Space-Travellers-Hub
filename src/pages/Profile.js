/* eslint-disable array-callback-return */
import React from 'react';
import { useSelector } from 'react-redux';

function Profile() {
  const rockets = useSelector((state) => state.rocketState);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved);
  return (
    <div className="profile">
      <h1>My Rockets</h1>
      { reservedRockets.map((rocket) => (
        <h4 className="profile-rockets" key={rocket.id}>{ rocket.name }</h4>
      )) }
    </div>
  );
}

export default Profile;
