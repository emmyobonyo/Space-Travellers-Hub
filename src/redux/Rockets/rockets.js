const initialState = [];
const FETCH_DATA = 'react-spacex/rockets/FETCH_DATA';

export const fetchRockets = () => async (dispatch) => {
  const response = await fetch('https://api.spacexdata.com/v3/rockets');
  const rockets = await response.json();
  const newRocket = [];
  rockets.forEach((rocket) => {
    newRocket.push({
      id: rocket.rocket_id,
      name: rocket.rocket_name,
      description: rocket.description,
      image: rocket.flickr_images,
    });
  });
  dispatch({ type: FETCH_DATA, newRocket });
};

export const fetchRocketsAction = (payload) => ({
  type: FETCH_DATA,
  payload,
});

const rocketsReducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_DATA:
      return action.newRocket;
    default:
      return state;
  }
};

export default rocketsReducer;
