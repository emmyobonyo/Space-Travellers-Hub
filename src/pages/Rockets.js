import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { fetchRockets } from '../redux/Rockets/rockets';

function Rockets() {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchRockets());
  }, []);

  return (
    <>
      <p>Hello World</p>
    </>
  );
}

export default Rockets;
