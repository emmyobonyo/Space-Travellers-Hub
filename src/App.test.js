/* eslint-disable no-unused-expressions */
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Navbar from './components/Navbar';
import Rockets from './pages/Rockets';
import Missions from './pages/Missions';
import Profile from './pages/Profile';
import store from './redux/configureStore';

it('should return the Navigation', () => {
  const navigation = render(<Router><Navbar /></Router>);
  expect(navigation).toMatchSnapshot();
});

it('should return the rockets page', () => {
  const rockets = render(<Provider store={store}><Rockets /></Provider>);
  expect(rockets).toMatchSnapshot();
  expect(rockets.queryByText(/Rockets/)).toBeInTheDocument;
});

it('should return the missions page', () => {
  const missions = render(<Provider store={store}><Missions /></Provider>);
  expect(missions).toMatchSnapshot();
  expect(missions.queryByText(/Missions/)).toBeInTheDocument;
});

it('should return the rockets page', () => {
  const profile = render(<Provider store={store}><Profile /></Provider>);
  expect(profile).toMatchSnapshot();
  expect(profile.queryByText(/My Missions/)).toBeInTheDocument;
});
