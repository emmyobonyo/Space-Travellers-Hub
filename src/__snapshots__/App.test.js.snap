// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`click events should return the missions page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="mission-table"
      >
        <table>
          <thead>
            <tr
              class="table-head"
            >
              <th>
                Mission
              </th>
              <th>
                Description
              </th>
              <th>
                Status
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="mission-table"
    >
      <table>
        <thead>
          <tr
            class="table-head"
          >
            <th>
              Mission
            </th>
            <th>
              Description
            </th>
            <th>
              Status
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`click events should return the profile page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <div
          class="profile__missions"
        >
          <h1
            class="profile__h1"
          >
            My Missions
          </h1>
          <ul
            class="profile__ul"
          />
        </div>
        <div
          class="profile"
        >
          <h1>
            My Rockets
          </h1>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <div
        class="profile__missions"
      >
        <h1
          class="profile__h1"
        >
          My Missions
        </h1>
        <ul
          class="profile__ul"
        />
      </div>
      <div
        class="profile"
      >
        <h1>
          My Rockets
        </h1>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`click events should return the rockets page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1
          class="rocket-heading"
        >
          All Rockets
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1
        class="rocket-heading"
      >
        All Rockets
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should return the Navigation 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="nav-bar"
      >
        <div
          class="image"
        >
          <img
            alt="Space-Ship Logo"
            class="img"
            src="planet.png"
          />
          <h3
            class="head"
          >
            Space Travelers' Hub
          </h3>
        </div>
        <ul
          class="list"
        >
          <li>
            <a
              aria-current="page"
              class="nav-link current"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              class="nav-link"
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li>
            <a
              class="nav-link"
              href="/myprofile"
            >
              My Profile
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="nav-bar"
    >
      <div
        class="image"
      >
        <img
          alt="Space-Ship Logo"
          class="img"
          src="planet.png"
        />
        <h3
          class="head"
        >
          Space Travelers' Hub
        </h3>
      </div>
      <ul
        class="list"
      >
        <li>
          <a
            aria-current="page"
            class="nav-link current"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            class="nav-link"
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li>
          <a
            class="nav-link"
            href="/myprofile"
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
